### PR TITLE
1.5 updates to Getting Started - Iterate on App page

### PR DESCRIPTION
**Pull request by corythorpe**
# Which other branches do you want a technical writer to cherry-pick this PR to (if any)?

It's best to PR to the most recent relevant branch and leave all the cherry-picking to the
writer, except where earlier branches require factual changes to the content.
For more information about the branches, see https://github.com/pivotal/docs-tap#branches

### CODE REVIEW

**Comment by snneji** [comment]
HI @corythorpe, is this ready to go yet? We only have a couple of weeks until 1.5 and I'll need time to edit your changes.

File: getting-started/iterate-new-app-intellij.hbs.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+   1. In the **Tiltfile Path** field, provide the path to the `Tiltfile` in the Tanzu Java Web App project directory. Select the folder icon on the right-side of the field, navigate to the Tanzu Java Web App directory, select the `Tiltfile` and click the `Open` button. The `Tiltfile` is used to facilitate Live Update using Tilt.
+   1. In the **Local Path** field, provide the path to the directory containing the Tanzu Java Web App. The Local Path value tells the Tanzu Developer Tools extension which directory on your local file system to bring into the source image container image.
+   For example, `/Users/developer/Documents/tanzu-java-web-app`.
+   1. In the **Source Image** field, provide the destination image repository to publish an image containing your workload source code. The Source Image value tells the Tanzu Developer Tools extension where to publish the container image with your uncompiled source code, and what to name that image. The image must be published to a container registry where you have write (push) access.
+   For example, `gcr.io/myteam/tanzu-java-web-app-source`.
+
```
Comment (paulcwarren) [inline getting-started/iterate-new-app-intellij.hbs.md:73]: question: are we missing an "apply" to save changes?

File: getting-started/iterate-new-app-intellij.hbs.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+   >For example, if you use docker consult [docker's docs](https://docs.docker.com/engine/reference/commandline/login/), if you use Harbor consult [Harbor's docs](https://goharbor.io/docs/1.10/working-with-projects/working-with-images/pulling-pushing-images/), etc.
+
+1. In the Project tab of IntelliJ, right-click the `Tiltfile` file under the application name `tanzu-java-web-app` and select `Run \'Tanzu Live Update - tanzu-java-web-app\'` to begin Live Updating the application on the cluster.
+1. Alternatively, select the `Edit Run/Debug configurations` dropdown in the top-right corner, select `Tanzu Live Update - tanzu-java-web-app`, then click the green play button to the right of the `Edit Run/Debug configurations` dropdown.
+The `Run` tab will open and display the output from Tanzu Application Platform and from Tilt indicating that the container is being built and deployed.
+    - On the `Tanzu Panel` tab you will see the status of Live Update reflected under the `tanzu-java-web-app` workload entry.
+    - Live update can take 1 to 3 minutes while the workload deploys and the Knative service becomes available.
```
Comment (paulcwarren) [inline getting-started/iterate-new-app-intellij.hbs.md:82]: 1-3 mins is a little loose I think.  depending on whether the deployed workload is up to date with your source code, or not.  If it is then live update can be started in ~10s

File: getting-started/iterate-new-app-intellij.hbs.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    >**Note** Depending on the type of cluster you use, you might see an error similar to the following:
+
+    >`ERROR: Stop! cluster-name might be production.
+    >If you're sure you want to deploy there, add:
+    >allow_k8s_contexts('cluster-name')
+    >to your Tiltfile. Otherwise, switch k8scontexts and restart Tilt.`
+    >Follow the instructions and add the line, `allow_k8s_contexts('cluster-name')` to your `Tiltfile`.
+
+2. When the Live Update task in the `Run` tab says that it was successful, resolve to "Live Update Started," use the hyperlink at the top of the Run output following the words **Tilt started on** to  view your application in your browser.
+3. In the IDE, make a change to the source code. For example, in `HelloController.java`, edit the string returned to say `Hello!` and save.
```
Comment (paulcwarren) [inline getting-started/iterate-new-app-intellij.hbs.md:93]: Then re-compile the changed class(es) or build the project

File: getting-started/iterate-new-app-intellij.hbs.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+4. The container is updated when the logs stop streaming. Navigate to your browser and refresh the page.
+5. View the changes to your workload running on the cluster.
+6. Either continue making changes, or stop and deactivate the live update when finished. To stop Live Update navigate to the `Run` tab at the bottom left of the IntelliJ window and click the red stop icon on the left side of the screen.
```
Comment (paulcwarren) [inline getting-started/iterate-new-app-intellij.hbs.md:96]: and deactivate?

File: getting-started/iterate-new-app-intellij.hbs.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+## <a id="debug-your-app"></a>Debug your application
+
+Debug the cluster either on the application or in your local environment.
+
+Use the following steps to debug the cluster:
+
+1. Set a breakpoint in your code.
+1. Create a Run Configuration.
```
Comment (paulcwarren) [inline getting-started/iterate-new-app-intellij.hbs.md:105]: First time only.

File: getting-started/iterate-new-app-intellij.hbs.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+The Apply Workload command will run, which opens a terminal and shows you the output of the Workload Apply. You can also monitor your application as it's being deployed to the cluster on the `Tanzu Panel`. The `Tanzu Panel` shows the workloads in the namespace associated with your current Kubernetes context on the left side, and the details of the Kubernetes resources for the workloads running in the namespace associated with your current Kubernetes context in the center. The Apply Workload command can take a few minutes to deploy your application onto the cluster.
+
+
+## <a id="live-update-your-app"></a>Live update your application
+
+Deploy the application to view it updating live on the cluster to see how code changes behave on a production cluster.
+
+Follow the following steps to live update your application:
+
+1. Create a Run Configuration.
```
Comment (paulcwarren) [inline getting-started/iterate-new-app-intellij.hbs.md:63]: First time only

File: getting-started/iterate-new-app-intellij.hbs.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+   For example, `gcr.io/myteam/tanzu-java-web-app-source`.
+
+   >**Note** Consult the documentation for the registry you're using to determine which steps are necessary to authenticate and gain push access.
+
+   >For example, if you use docker consult [docker's docs](https://docs.docker.com/engine/reference/commandline/login/), if you use Harbor consult [Harbor's docs](https://goharbor.io/docs/1.10/working-with-projects/working-with-images/pulling-pushing-images/), etc.
+
+   8. In the **Namespace** field, optionally provide the namespace that the workload should be associated with on the cluster. If you followed the steps to [Prepare your IDE to iterate on your application](/iterate-new-app-intellij.hbs.md#prepare-to-iterate) you do not need to enter a namespace and IntelliJ will use the namespace you associated with your context.
```
Comment (paulcwarren) [inline getting-started/iterate-new-app-intellij.hbs.md:120]: Apply the changes?

File: getting-started/iterate-new-app-intellij.hbs.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+The `Run` tab will open and display the output from Tanzu Application Platform and from Tilt indicating that the container is being built and deployed.
+    - On the `Tanzu Panel` tab you will see the status of Live Update reflected under the `tanzu-java-web-app` workload entry.
+    - Live update can take 1 to 3 minutes while the workload deploys and the Knative service becomes available.
+
+1. In the Explorer tab of IntelliJ, right-click any file under the application name `tanzu-java-web-app` and select `Tanzu: Java Debug Start` to begin debugging your application on the cluster. In a few moments the workload is redeployed with debugging enabled. The `Debug` tab will show the **Connected to the target VM, address: 'localhost:** followed by a port number, indicating that the debugger has attached.
+1. Navigate to `http://localhost:<port number>` in your browser, where <port number> is the number found after `localhost` in the `Debug` tab. This hits the breakpoint within IntelliJ. Play to the end of the debug session using IntelliJ debugging controls.
```
Comment (paulcwarren) [inline getting-started/iterate-new-app-intellij.hbs.md:129]: Kinda...the User has to use the application to cause the breakpoint to be hit.  That's a little different to just visiting the first page of the application.  That works for our sample app perhaps.  But wont work always.  It depends where the breakpoint is set.

File: getting-started/iterate-new-app-intellij.hbs.md (hunk 1 above)
Comment (corythorpe) [reply getting-started/iterate-new-app-intellij.hbs.md:73]: good catch

File: getting-started/iterate-new-app-intellij.hbs.md (hunk 6 above)
Comment (corythorpe) [reply getting-started/iterate-new-app-intellij.hbs.md:63]: Good point. This content is in the getting started guide, so I think we're safe to assume someone is doing all of this for the first time. If we use any of this in our general help docs we should keep this in mind.

File: getting-started/iterate-new-app-intellij.hbs.md (hunk 4 above)
Comment (corythorpe) [reply getting-started/iterate-new-app-intellij.hbs.md:96]: great catch, no idea where this came from, it must be a relic from MVP times

File: getting-started/iterate-new-app-intellij.hbs.md (hunk 7 above)
Comment (corythorpe) [reply getting-started/iterate-new-app-intellij.hbs.md:120]: good catch, looks like i missed this in all of our run configs

File: getting-started/iterate-new-app-intellij.hbs.md [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+1. Alternatively, select the `Edit Run/Debug configurations` dropdown in the top-right corner, select `Tanzu Debug Workload - tanzu-java-web-app`, then click the green debug button to the right of the `Edit Run/Debug configurations` dropdown.
+The `Run` tab will open and display the output from Tanzu Application Platform and from Tilt indicating that the container is being built and deployed.
+    - On the `Tanzu Panel` tab you will see the status of Live Update reflected under the `tanzu-java-web-app` workload entry.
+    - Live update can take 1 to 3 minutes while the workload deploys and the Knative service becomes available.
+
+1. In the Explorer tab of IntelliJ, right-click any file under the application name `tanzu-java-web-app` and select `Tanzu: Java Debug Start` to begin debugging your application on the cluster. In a few moments the workload is redeployed with debugging enabled. The `Debug` tab will show the **Connected to the target VM, address: 'localhost:** followed by a port number, indicating that the debugger has attached.
```
[the remark this answers is not in the document]
Comment (corythorpe) [reply getting-started/iterate-new-app-intellij.hbs.md:128]: this is a duplicate, and the entry above erroneously mentions live update for some reason

**Comment by corythorpe** [comment]
Conflict can safely be resolved as it unrelated to my changes.

**Pull request merged**